### DISCUSSION TITLE
Label application flow usability issues

### DISCUSSION THREAD

**Discussion by peppy** [community/community#42] [Mobile]
I spend some minutes to hours each day keeping up with notifications and organising issues across many repositories. Assigning labels is quite important, and I still prefer to use the github web site (even on iPad) due to some usability issues in the app. The main ones currently are:

- You need to tap the search bar before typing text to filter labels. I'd prefer it selects automatically.
- Kind of related, but some keys pass through the dialog and perform unexpected actions on the main app view (try 'u' for example)
- Search is unreliable – we use namespaced labels such as `area:song-select`. Serching for "song" results in no matches. Searching for "area:so" results in no matches.
- There's no keyboard shortcut to exit-and-save the dialog. On web you can hit `esc` for this.

**Comment by candyho** [comment]
Hey thanks for sharing your feedback! We acknowledge the in-app search experience could be improved and that is something on our radar.

Some questions regarding to the scenarios you've described: 

> - You need to tap the search bar before typing text to filter labels. I'd prefer it selects automatically.

Do you mean you'd like to have an UI element (like a button) where you can just tap and filter issues by labels instead of typing text in the search bar?

> - There's no keyboard shortcut to exit-and-save the dialog. On web you can hit `esc` for this.

Just confirming, you're expecting to use the same keyboard shortcut on iPad right?

**Reply by peppy** [reply]
Sorry I may not have been clear enough. This is specifically for *adding* labels, not searching by them. I'm talking about the search textbox that is on the add/modify labels modal dialog:

<img width="1180" alt="ppyosu-web  username blocks input on user card #6844" src="https://user-images.githubusercontent.com/191335/97943914-f15c7d80-1dc7-11eb-9ba2-0addd0996184.png">

My workflow is mostly keyboard based, so optimally i would like to be able to:

`j` `j` `j`  (select notification to work on from notification list view)
`l` (bring up label dialog)
`code-qu` (searching for label)
`enter` (add label)
`cmd-a` (select search text)
`UX` (searching for second label)
`enter` (add second label)
`escape` (exit label dialog and apply changes)

The keyboard flow is how it works on the web site at the moment, for what it's worth.

**Reply by candyho** [reply]
Thanks for the explanation. Sorry for mistaking label specific search for generic in-app search. It all makes sense. I will surface your feedback to the team and see if we can do some fixes here.

**Comment by rnystrom** [comment]
> You need to tap the search bar before typing text to filter labels. I'd prefer it selects automatically.

I tried this out on the phone, and I think it's a lot of motion. Have you tried the <kbd>⌘</kbd><kbd>f</kbd> shortcut to quickly focus?

> There's no keyboard shortcut to exit-and-save the dialog. On web you can hit esc for this.

Have you used <kbd>⌘</kbd><kbd>enter</kbd>? That should save & dismiss. The escape shortcut triggers a "cancel".

> Kind of related, but some keys pass through the dialog and perform unexpected actions on the main app view

Which version if iPadOS are you using?

**Reply by peppy** [reply]
> I tried this out on the phone, and I think it's a lot of motion. Have you tried the ⌘f shortcut to quickly focus?

Aha, cmd-f does seem to focus! But the next steps are a bit weird (selecting a matching label) – even after hitting enter to unfocus the textbox, using up/down j/k doesn't seem to allow selecting the label (in the case there's only one). Even when there's more than one matching result, having to press enter first is a bit hard to process mentally.

Also to clarify, my original proposal was to only have the search field focus after starting to type on an external keyboard (not on initial display). But can understand if that's not something which works from a UX perspective.

> Have you used ⌘enter? That should save & dismiss. The escape shortcut triggers a "cancel".

This works! Now that I know it exists this should be enough to get by with, thanks.

> Which version if iPadOS are you using?

iPadOS 14.2 (GM i guess?)

Here's a video, note the small popup at the bottom of the screen as i press 'U':  https://youtu.be/DM20Hp3cWNU

**Reply by rnystrom** [reply]
I've seen reports about key command conflicts with other surfaces on 14.2, thanks for confirming. I'll take a look!

I like the ideas around auto-focusing keyboard on typing, will give that some thought as well.

**Reply by rnystrom** [reply]
Following up that we have a fix up internally for "background" keyboard shortcuts being activated. Should be out soon!
